tests/unit/cmorizers/obs/test_utilities.py

```diff
@@ -101,3 +109,206 @@ def test_fix_dtype_not_lazy(cube):
         if coord.has_bounds():
             assert coord.bounds_dtype == np.float64
     assert not is_lazy(cube)
+
+
+def VarInfoObj(var_dict):
+    mock_dict = Mock()
+    mock_dict.__dict__ = var_dict
+    return mock_dict
+
+
+def _create_sample_cube():
+    """Create a quick CMOR-compliant sample cube."""
+    coord_sys = iris.coord_systems.GeogCS(iris.fileformats.pp.EARTH_RADIUS)
+    cube_data = np.ones((2, 3, 2, 2))
+    cube_data[1, 1, 1, 1] = 22.
+    time = iris.coords.DimCoord([15, 45],
+                                standard_name='time',
+                                bounds=[[1., 30.], [30., 60.]],
+                                units=Unit('days since 1950-01-01',
+                                           calendar='gregorian'))
+    zcoord = iris.coords.DimCoord([0.5, 5., 50.],
+                                  var_name='depth',
+                                  standard_name='depth',
+                                  bounds=[[0., 2.5], [2.5, 25.], [25., 250.]],
+                                  units='m',
+                                  attributes={'positive': 'down'})
+    lons = iris.coords.DimCoord([1.5, 2.5],
+                                standard_name='longitude',
+                                bounds=[[1., 2.], [2., 3.]],
+                                units='degrees_east',
+                                coord_system=coord_sys)
+    lats = iris.coords.DimCoord([1.5, 2.5],
+                                standard_name='latitude',
+                                bounds=[[1., 2.], [2., 3.]],
+                                units='degrees_north',
+                                coord_system=coord_sys)
+    coords_spec = [(time, 0), (zcoord, 1), (lats, 2), (lons, 3)]
+    cube = iris.cube.Cube(cube_data, dim_coords_and_dims=coords_spec)
+    return cube
+
+
+def test_add_scalar_height_coord():
+    """Test add height aux coord."""
+    cube = _create_sample_cube()
+    utils.add_scalar_height_coord(cube, height=10.)
+    assert cube.coord("height").points[0] == 10.
+    assert "positive" in cube.coord("height").attributes
+    assert cube.coord("height").attributes["positive"] == "up"
+
+
+def test_convert_time_units():
```

Comment:
maybe you could use `pytest.mark.parametrize` to loop over all cases?

Reply:
good call - done! :beer: